src/Components/GEOSldas_GridComp/Shared/LDAS_ensdrv_Globals.F90

```diff
@@ -141,7 +143,33 @@ elemental   function LDAS_is_nodata(data) result(no_data)
   end function LDAS_is_nodata
 
   ! *************************************************************
-  
+  !
+  ! return ensemble id string "_eXXXX"
+
+  subroutine get_ensid_string(ensid_string, id, ens_id_width, num_ensemble)
+
+     character(*), intent(inout) :: ensid_string
+     integer,      intent(in)    :: id
+     integer,      intent(in)    :: ens_id_width
+     integer,      intent(in)    :: num_ensemble
+
+     character(len=ESMF_MAXSTR)  :: fmt_str
+
+     if (num_ensemble == 1) then
+        ensid_string = ''
+        return
+     endif
+     
+     if (ens_id_width <=0) then
+       call ldas_abort(LDAS_GENERIC_ERROR, 'get_ensid_string', 'ens_id_width should be greater than 1')
+     endif
+
+     write (fmt_str, "(A2,I1,A1,I1,A1)") "(I", ens_id_width,".",ens_id_width,")"
```

Comment:
I think this formatting will fail for ens_id_width>9 because of the "I1" format.  I don't think we need to worry about ens_id_width larger than 9, so I think we should just restrict ens_id_width to be less than 10.
While we're at it, I think we should have ens_id_width be a minimum of 3, then the issue with the number of digits for the LADAS ensemble met forcing coupling being hardwired to 3 is also taken care of. 
So maybe add something like the following (maybe here, or perhaps better where we first read NUM_ENSEMBLE and ens_id_width)
```
if (num_ensemble>) then
_ASSERT(ens_id_width>2 .and. ens_id_width<10,"Must use 3<=ens_id_width<=9 for NUM_ENSEMBLE>1")
end if
```
Does this make sense?